tests/utils/log/test_logging.py

```diff
@@ -21,72 +21,84 @@
 class TestS3TaskHandler(unittest.TestCase):
 
     def setUp(self):
-        super(S3TaskHandler, self).setUp()
+        super(TestS3TaskHandler, self).setUp()
         self.remote_log_location = 'remote/log/location'
+        self.local_log_location = 'local/log/location'
+        self.s3_log_location = 's3/log/location'
+        self.filename_template = ''
         self.hook_patcher = mock.patch("airflow.hooks.S3_hook.S3Hook")
         self.hook_mock = self.hook_patcher.start()
         self.hook_inst_mock = self.hook_mock.return_value
         self.hook_key_mock = self.hook_inst_mock.get_key.return_value
         self.hook_key_mock.get_contents_as_string.return_value.decode.\
             return_value = 'content'
+        class UnitTestS3TaskHandler(S3TaskHandler):
+            def test_instance(self):
+                return True
+        self.s3_task_handler = UnitTestS3TaskHandler(
+            self.local_log_location,
+            self.s3_log_location,
+            self.filename_template
+        )
 
     def tearDown(self):
         self.hook_patcher.stop()
-        super(S3TaskHandler, self).tearDown()
+        super(TestS3TaskHandler, self).tearDown()
 
     def test_init(self):
-        S3TaskHandler()
+        self.s3_task_handler.hook()
         self.hook_mock.assert_called_once_with('')
 
     def test_init_raises(self):
         self.hook_mock.side_effect = Exception('Failed to connect')
-        handler = S3TaskHandler()
+        handler = self.s3_task_handler
         with mock.patch.object(handler.log, 'error') as mock_error:
             # Initialize the hook
-            handler.hook()
+            handler.hook
```

Comment:
Is this change correct? It looks suspect.

Reply:
The hook method in the handler is now a property, hence the change here!